grai-integrations/source-snowflake/src/grai_source_snowflake/loader.py

```diff
@@ -0,0 +1,188 @@
+import os
+from itertools import chain
+from typing import Any, Callable, Dict, List, Optional, Union
+import snowflake.connector
+from grai_source_snowflake.models import (
+    Column,
+    ColumnID,
+    Edge,
+    EdgeQuery,
+    SnowflakeNode,
+    Table,
+)
+
+
+def get_from_env(label: str, default: Optional[Any] = None, validator: Callable = None):
+    env_key = f"GRAI_SNOWFLAKE_{label.upper()}"
+    result = os.getenv(env_key, default)
+    if result is None:
+        message = (
+            f"Establishing snowflake connection requires a {label}. "
+            f"Either pass a value explicitly or set a {env_key} environment value."
+        )
+        raise Exception(message)
+    return result if validator is None else validator(result)
+
+
+class SnowflakeConnector:
+    def __init__(
+        self,
+        account: Optional[str] = None,
+        user: Optional[str] = None,
+        password: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        role: Optional[str] = None,
+        database: Optional[str] = None,
+        schema: Optional[str] = None,
+        namespace: Optional[str] = None,
+    ):
+        self.account = account 
```

Comment:
All of the connection parameters on the connector are currently `Optional`, I can't directly test but looking at the connection dict it seems as if they are all actively required. If so, I'd switch the typing and remove the default value or leave them optional and handle things the way as namespace is currently handled with a fallback to environment settings.